exercise_1/main.cpp

```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include <fstream>
+
+const int V=8;
+
+int *Dijkstra(int **GR, int V, int st) {
+    int *distance, count, index, i, u;
+    bool *visited;
+    distance = new int [V];
+    visited = new bool [V];
+    for (i=0; i<V; i++) { distance[i]=INT32_MAX; visited[i]=false; }
+    distance[st]=0;
+    for (count=0; count<V-1; count++) {
+        int min=INT32_MAX;
+        for (i=0; i<V; i++)
+            if (!visited[i] && distance[i]<=min) { min=distance[i]; index=i; }
+        u=index;
+        visited[u]=true;
+        for (i=0; i<V; i++)
+            if (!visited[i] && GR[u][i] && distance[u]!=INT32_MAX &&
+                distance[u]+GR[u][i]<distance[i])
+                distance[i]=distance[u]+GR[u][i];
+    }
+    return distance;
+}
+
+int main(int argc, char **argv) {
+    int start, **GR;
+    GR = new int * [V];
```

Comment:
please use std:: containers instead. 
the reason behind this suggestion is that containers are almost as good as raw structures performance-wise but they offer you far greater safety.